src/Tasks/DownloadFile.cs

```diff
@@ -148,9 +148,13 @@ private async Task DownloadAsync(Uri uri, CancellationToken cancellationToken)
                     }
                     catch (HttpRequestException e)
                     {
-                        // HttpRequestException does not have the status code so its wrapped and thrown here so that later on we can determine
-                        // if a retry is possible based on the status code
+#if RUNTIME_TYPE_NETCORE
+                        // MSBuild History: CustomHttpRequestException was created as a wrapper over HttpRequestException
+                        // so it could include the StatusCode. As of net5.0, the statuscode is now in HttpRequestException.
+                        throw new HttpRequestException(e.Message, e.InnerException, response.StatusCode);
```

Comment:
Replace this with just `throw;`? You're losing the stack and wasting a little memory (admittedly not in a hot path).

Reply:
I'd rather keep things as they are and keep as much info when something bad goes wrong. This PR is about changing TF so there shouldn't be much functional change.

Reply:
You're _losing_ information by wrapping it in a new HttpRequestException. When new functionality becomes available to make logging errors nicer, I'd do it.

Reply:
Ah I misread your original post as "you're losing the stack by using `throw;`."

Not happy with how it'll look (stylistically) so any suggestions are appreciated.

the NET6_0_etcetc condition needs to avoid a `e is declared but not being used` warning that'll turn into an error in official builds.
```
try
{​
    response.EnsureSuccessStatusCode();
}​
#if NET6_0_OR_GREATER
catch (HttpRequestException)
{​
   throw;
#else
catch (HttpRequestException e)
{​
    // MSBuild History: CustomHttpRequestException was created as a wrapper over HttpRequestException
    // so it could include the StatusCode. As of net5.0, the statuscode is now in HttpRequestException.
    throw new CustomHttpRequestException(e.Message, e.InnerException, response.StatusCode);
#endif
}​
```

Reply:
#if !NET6_0_OR_GREATER
try
{
#endif
response.EnsureSuccessStatusCode();
#if !NET6_0_OR_GREATER
}
catch (HttpRequestException e)
{
// comment
throw new...
}
#endif